### PR TITLE
Allow more Timezone formats for "Instant" in REST API

### LINKED ISSUE #2223: Issue with Timezones when writing timestamp data type
When inserting timestamp data to AstraDB via Stargate REST API, the only timezone that works is Z.
No other timezones or timezone offsets work. Here are the ones I've tested:

**Working:**
- 2022-11-08T10:30:00Z
- 2022-11-08T10:30:00.000Z

**Not Working**
- Bad request: Invalid Timestamp value: Text '2022-11-08T10:30:00.000' could not be parsed at index 23
- Bad request: Invalid Timestamp value: Text '2022-11-08T10:30:00.000+11' could not be parsed at index 23
- Bad request: Invalid Timestamp value: Text '2022-11-08T10:30:00.000+1100' could not be parsed at index 23
- Bad request: Invalid Timestamp value: Text '2022-11-08T10:30:00.000+11:00' could not be parsed at index 23
- Bad request: Invalid Timestamp value: Text '2022-11-08T10:30:00' could not be parsed at index 19
- Bad request: Invalid Timestamp value: Text '2022-11-08T10:30:00+11' could not be parsed at index 19
- Bad request: Invalid Timestamp value: Text '2022-11-08T10:30:00+1100' could not be parsed at index 19
- Bad request: Invalid Timestamp value: Text '2022-11-08T10:30:00+11:00' could not be parsed at index 19

### Test Environment

**Table definition**
```
CREATE TABLE demo.sendemail (
    timestamp timestamp PRIMARY KEY,
    emaildata text
)
```

**API POST**
```
curl --location --request POST 'https://1b44200e-93df-4b4e-b250-005c20557cde-australia-southeast1.apps.astra.datastax.com/api/rest/v2/keyspaces/demo/sendemail' \
--header 'X-Cassandra-Token: <token>' \
--header 'Content-Type: application/json' \
--data-raw '{
    "timestamp": "2022-11-08T10:30:00.000",
    "emaildata": "this is my mail message"
}'
```

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Improves parsing of `Timestamp` values to allow for more options; basically `Z`, 2 and 4 digit offsets, with or without colon.

**Which issue(s) this PR fixes**:
Fixes #2223

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java [hunk 1]
```diff
@@ -599,7 +611,11 @@ public QueryOuterClass.Value protoValueFromStringified(String value) {
         // TODO: this implementation requires full date/time specification including timezone
         // we could support more flexibility in format as requested in
         // https://github.com/stargate/stargate/issues/839
```
Comment (ivansenic) [inline apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java:613]: should the TODO be removed now as well?

File: apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java [hunk 2]
```diff
@@ -21,6 +22,17 @@
...
 public class ToProtoValueCodecs {
+  /**
+   * Ok, this formatter is combination of info from:
+   *
+   * <ul>
+   *   <li>https://stackoverflow.com/questions/43360852/cannot-parse-string-in-iso-8601-format-lacking-colon-in-offset-to-java-8-date
+   *   <li>https://stackoverflow.com/questions/34637626/java-datetimeformatter-for-time-zone-with-an-optional-colon-separator
+   * </ul>
+   */
+  private static final DateTimeFormatter ISO_OFFSET_DATE_TIME_OPTIONAL_COLON =
```
Comment (ivansenic) [inline apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java:33]: would be nice to have in the javadocs the formats that would be supported by this.. I guess `[]` means optional, and while `[.SSS]` is clear, the two different options using the `X` is confusing for me?

**Review by ivansenic — APPROVED** [review]
Cool, did we cover now everything we need to?

File: apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java (hunk 1 above)
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java:613]: Yes, I'll do that.

**Comment by tatu-at-datastax** [comment]
> Cool, did we cover now everything we need to?

I think so, wrt implementation of textual representations, matching unit tests for converters (for integration test also added some of formats but not all).

File: apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java (hunk 2 above)
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java:33]: Ok I can add it. As I understand it, `X` variants are for with-colon (`08:00`) as well as without (`0800`). unfortunately even `DateTimeFormat` Javadocs are bit confusing on what exactly is allowed; SO answer gives bit more practical answer.

**Comment by HiltonRosenfeld** [comment]
This is terrific!
When should this be functional in Astra serverless (via stargate)?

**Pull request merged**